src/urllib3/util/url.py

```diff
@@ -413,8 +413,8 @@ def parse_url(url: str) -> Url:
         if normalize_uri and fragment:
             fragment = _encode_invalid_chars(fragment, _FRAGMENT_CHARS)
 
-    except (ValueError, AttributeError):
-        raise LocationParseError(source_url) from None
+    except (ValueError, AttributeError) as parse_exc:
```

Comment:
I think this makes sense, could we change this to `as e:` to match elsewhere in the codebase?

Also could you give an example backtrace of what this would look like in action?

Reply:
```python3
>>> urllib3.util.url.parse_url('http://foo:1000000000')
Traceback (most recent call last):
  File "/usr/lib/python3/dist-packages/urllib3/util/url.py", line 368, in parse_url
    auth, host, port = SUBAUTHORITY_RE.match(authority).groups()
AttributeError: 'NoneType' object has no attribute 'groups'

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/usr/lib/python3/dist-packages/urllib3/util/url.py", line 392, in parse_url
    raise LocationParseError(source_url) from parse_exc
urllib3.exceptions.LocationParseError: Failed to parse: http://foo:1000000000
>>> 
```

Reply:
(requested change made in 72ee1db)